api/v1alpha1/propagation_types.go

```diff
@@ -155,15 +157,43 @@ type DeploymentStatusesReport struct {
 	Statuses       []DeploymentStatus `json:"statuses,omitempty"`
 }
 
+func (r *DeploymentStatusesReport) VersionHealthyDuration(version string, at time.Time) time.Duration {
+	for i, s1 := range r.Statuses {
+		if s1.Version != version {
+			continue
+		}
+		until := at
+		for _, s2 := range r.Statuses[i:] {
+			if s2.State != HealthStateHealthy {
+				if s2.Version == s1.Version {
+					return 0
+				} else {
+					until = s2.Start.Time
```

Comment:
I think only if `until > s2.Start.Time`.
The arguments list for the function assumes I can ask "For how long this version has been healthy at specific time?". Imagine an version was deployed at t-5h and became healthy, but then became unhealthy at t-2h. And now I'm asking "for how long it was healthy at = t-3h?". Correct answer on tis question would be "2h", but the code will return "3h"